### PR TITLE
feat: impl `dns.getServers`

### PR DESCRIPTION
**Pull request by Hanaasagi**
Close: #3981

### What does this PR do?

- Implement the `dns.getServers` function

Reference:

- https://nodejs.org/api/dns.html#dnsgetservers
- https://c-ares.org/ares_get_servers_ports.html

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote automated tests

- [x] I ran `make js` and committed the transpiled changes
- [x] I or my editor ran Prettier on the changed files (or I ran `bun fmt`)
- [x] I included a test for the new code, or an existing test covers it

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I or my editor ran `zig fmt` on the changed files
- [x] I included a test for the new code, or an existing test covers it
- [ ] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
nice

File: test/js/node/dns/node-dns.test.js [hunk 1]
```diff
@@ -190,3 +192,34 @@ test("dns.lookup (localhost)", done => {
     done(err);
   });
 });
+
+test("dns.getServers", done => {
+  function parseResolvConf() {
+    let servers = [];
+    try {
+      const content = fs.readFileSync("/etc/resolv.conf", "utf-8");
```
Comment (Jarred-Sumner) [inline test/js/node/dns/node-dns.test.js:200]: I think this test will only work on linux 

can use test.skipIf(process.platform === “darwin”)

**Review by Jarred-Sumner — APPROVED** [review]
Thank you. This looks good. Let’s let the tests run. 

Only part I’m not sure about is the buffer length. If any of the numbers are not 100% correct in all the cases it could cause a crash or undefined behavior and it’d be hard to reproduce. 

Assuming the current length isn’t huge (like 64 bytes probably), I suggest adding an extra 512 bytes to it just for wiggle room

**Comment by Hanaasagi** [comment]
About the buffer length, I checked the source code of `ares_inet_ntop`. It checks the `size` parameter we pass in, so overflow should not occur. However, I think I should handle the case where the return value is `NULL`.

https://github.com/c-ares/c-ares/blob/bad62225b7f6b278b92e8e85a255600b629ef517/src/lib/inet_ntop.c#L58-L95

And `std.fmt.bufPrint` will also check buf length.


https://github.com/ziglang/zig/blob/fc6e5756848cd627b69756853937ce22500e62cc/lib/std/fmt.zig#L1996
https://github.com/ziglang/zig/blob/fc6e5756848cd627b69756853937ce22500e62cc/lib/std/io/fixed_buffer_stream.zig#L63

**Comment by Jarred-Sumner** [comment]
> About the buffer length, I checked the source code of `ares_inet_ntop`. It checks the `size` parameter we pass in, so overflow should not occur. However, I think I should handle the case where the return value is `NULL`.
> 
> https://github.com/c-ares/c-ares/blob/bad62225b7f6b278b92e8e85a255600b629ef517/src/lib/inet_ntop.c#L58-L95
> 
> And `std.fmt.bufPrint` will also check buf length.
> 
> https://github.com/ziglang/zig/blob/fc6e5756848cd627b69756853937ce22500e62cc/lib/std/fmt.zig#L1996 https://github.com/ziglang/zig/blob/fc6e5756848cd627b69756853937ce22500e62cc/lib/std/io/fixed_buffer_stream.zig#L63

👍

**Pull request merged**